framework/Optimizers/GradientBasedOptimizer.py

```diff
@@ -73,6 +73,24 @@ def __init__(self):
     self.counter['varsUpdate'      ] = {}
     self.counter['solutionUpdate'  ] = {}
     self.counter['lastStepSize'    ] = {}              # counter to track the last step size taken, by trajectory
+    self.counter['iSave']            = {}              # integer work array of dimension 2 for line search
```

Comment:
please, explain a bit better here. What is it used for? The shape of the array? etc?

This comment is valid for all the entries below